.github/workflows/cd-circt.yml

```diff
@@ -1,11 +1,316 @@
+            --from=markdown
+            --wrap=none
+            -o metadata/pr-body.md
+            /dev/null
+      - name: Generate Commit Message
+        uses: docker://pandoc/core
+        with:
+          args: >-
+            --metadata-file=metadata/metadata.json
+            --template chisel/.github/workflows/cd-circt/commit-message.md
+            --from=markdown
+            -o metadata/commit-message.md
+            /dev/null
+      # Print the populated pandoc templates to the GitHub Actions step summary.
+      # (These will print on the "Summary" of a run of this Action.)
+      - name: Show Metadata
+        run: |
+          echo "# PR Title" >> $GITHUB_STEP_SUMMARY
+          cat metadata/pr-title.md >> $GITHUB_STEP_SUMMARY
+
+          echo "# PR Body" >> $GITHUB_STEP_SUMMARY
+          cat metadata/pr-body.md >> $GITHUB_STEP_SUMMARY
+
+          echo "# Commit Message" >> $GITHUB_STEP_SUMMARY
+          cat metadata/commit-message.md >> $GITHUB_STEP_SUMMARY
+      # Using the information in the JSON Metadata, cherry-pick all commits from
+      # the `ci/ci-circt-nightly` branch.
+      - name: Cherry Pick Staging Branch Commits
+        run: |
+          cd chisel/
+          git checkout main
+
+          git config user.name ${{ env.user }}
+          git config user.email ${{ env.email }}
+
+          COMMITS=$(jq -r '.commits[].checksum' < ../metadata/metadata.json)
+          for commit in $COMMITS; do
+            git cherry-pick $commit
+          done
+      # Create unstaged changes that updates the CIRCT version to "latest".
+      - name: Update Chisel to Latest CIRCT
+        run: |
+          cd chisel/
+
+          jq '.version = "${{ needs.determine-versions.outputs.version-latest }}"' < ${{ env.circt-config }} > ../metadata/tmp.json
+          mv ../metadata/tmp.json ${{ env.circt-config }}
+      # Put the pandoc-created PR title and commit message into step outputs so
+      # that we can pass these to the peter-evans/create-pull-request composite
+      # action in the next step.  Pass the commit message using a HereDoc (as
+      # GitHub suggests [4]) so that this will show up in a multi-line format
+      # correctly.  Use a unique HereDoc delimiter to avoid collisions with
+      # anything which may be in the commit message.
+      #
+      # The PR body is passed as a file in the next step because the
+      # peter-evans/create-pull-request composite action supports it.
+      #
+      # [4]: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+      - name: Setup Environment with Metadata
+        id: setup-env
+        run: |
+          echo pr-title=$(cat metadata/pr-title.md) >> $GITHUB_OUTPUT
+
+          EOF=$(cat /dev/urandom | base64 -w 128 | head -n8 | tail -n1)
+          {
+            echo "commit-message<<$EOF"
+            cat metadata/commit-message.md
+            echo $EOF
+          } >> $GITHUB_OUTPUT
```

Comment:
Another option could be to create the commit ourselves and use `git commit -F`. But yeah this is reasonable. I wish the action had a `commit-message-path` option 🙃

Reply:
It would be easy enough to fix this on the upstream repo. I'll leave this for now.